core/src/main/scala/com/raphtory/components/querymanager/AnalysisMessages.scala

```diff
@@ -69,8 +69,23 @@ case object RecheckTime                 extends QueryManagement
 case object RecheckEarliestTime         extends QueryManagement
 case class CheckMessages(jobId: String) extends QueryManagement
 
-case class VertexMessage[+T](superstep: Int, vertexId: Long, data: T) extends QueryManagement
-case class VertexMessageBatch[T](data: Array[VertexMessage[T]])       extends QueryManagement
+sealed trait GenericVertexMessage extends QueryManagement {
+  def superstep: Int
+  def vertexId: Long
+}
+
+case class VertexMessage[+T](superstep: Int, vertexId: Long, data: T) extends GenericVertexMessage
+
+case class VertexMessageBatch(data: Array[GenericVertexMessage]) extends QueryManagement
+
+case class FilteredEdgeMessage(superstep: Int, vertexId: Long, sourceId: Long)
```

Comment:
Is this needed when we have In and Out? or is this for both In/Out?

Reply:
Can't see it being used, but I could be totally blind.

Reply:
I think we put this in initially but then didn't use it. Maybe we should actually use it as we could reduce the number of messages sent.

Reply:
Ok cool, we can revisit this then when we do the expansion of the filtering you mention in the PR notes